src/glmfit.jl

```diff
@@ -402,7 +405,7 @@ function _fit!(m::AbstractGLM, verbose::Bool, maxiter::Integer, minstepfac::Real
         ## If the deviance isn't declining then half the step size
         ## The rtol*dev term is to avoid failure when deviance
         ## is unchanged except for rouding errors.
-        while dev > devold + rtol*dev
+        while !isfinite(dev) || dev > devold + rtol*dev
```

Comment:
It seems we could possibly end up in infinite loop here?

Reply:
Doesn't line 410 guard against that?